WIP: output mostly lus for walk script

psimyn opened the pull request.

psimyn committed:
output mostly lus for walk script

Beherith commented:
Why did you close this PR? Its a feature that is sorely needed in the future :)

psimyn committed:
Cleanup

psimyn commented:
I planned to recreate once I had cleaned up code a bit. Wanted to think of a way to make it work for both that isn't just "copy the entire block and rename to outputLua"

psimyn committed:
remove bMoving from walk loop

rename it to walking and use to prevent duplicate Walking threads
Also moves Signal and SetSignalMask together

psimyn committed:
Re-add walking check for walk loop

Will probably readd it to each keyframe as well, there were situations where
unit could get stuck animated running on the spot (i.e. jump to unpathable terrain)

psimyn commented:
remaining work:

1. do we need the `if bMoving` for each keyframe? I re-added it to the main walking loop. I assume that means that may take one extra step before stopping
1. animation for non is_walking not finished (mainly needs indentation fix)
1. death animations not working yet
1. adding separate option/button for LUS output. Might just copy the entire code block and have a save button for each filetype

psimyn committed:
Convert to radians to save runtime math.rad calls 

Definitely broken for non-walk and death scripts

psimyn committed:
Add separate button to output LUS

psimyn committed:
separate speed and sleeptime params

Beherith commented:
Scream if you are ready :)

Beherith commented:
> 
> 
> I planned to recreate once I had cleaned up code a bit. Wanted to think of a way to make it work for both that isn't just "copy the entire block and rename to outputLua"

I give zero fox about code quality, just that it works, which is probably pretty obvious from the existing code.

psimyn committed:
Split tobos into keyframe creation and file writer

Bring back BOS export, though it still seems to be broken somehow

psimyn committed:
Make BOS output work again

I just rearranged the order in register(), no idea
what's going on there

psimyn committed:
Fix copypaste bug with unit formatting

psimyn commented:
@Beherith I think this now mostly works. I might have messed up some stuff with linebreaks and indents - switched to macbook today and some things were looking pretty weird.

I still haven't finished all the death/nonwalk animation bits, but at least now BOS is not broken (assuming my small changes to `tobos` worked).

psimyn committed:
Make sure output always uses tabs

For some reason even tab characters were getting converted to spaces
though this might just have been my editor
